src/Plutarch/Test/Precompiled.hs

```diff
@@ -0,0 +1,179 @@
+{-# LANGUAGE RankNTypes #-}
+{-# LANGUAGE RecordWildCards #-}
+
+{- | Module: Plutarch.Test.Precompiled
+ Copyright: (C) Liqwid Labs 2022
+ Maintainer: Seungheon Oh <seungheon@mlabs.city>
+ Portability: GHC only
+ Stability: Experimental
+
+ Helpers for building test tree that shares same precompiled script.
+-}
+module Plutarch.Test.Precompiled (
+    Expectation (..),
+    TestCase,
+    (@&),
+    withApplied,
+    TestCompiled (..),
```

Comment:
Do people need to see the constructor here? I think our choice of monadic stack is very much an implementation detail nobody needs to see.